doc/internal/man3/OSSL_METHOD_STORE.pod

```diff
@@ -102,6 +102,12 @@ The I<method_up_ref> function is called to increment the
 reference count of the method and the I<method_destruct> function is called
 to decrement it.
 
+=head1 NOTES
+
+The I<prop_query> argument to ossl_method_store_cache_get() and
```

Comment:
This just seems wrong to me conceptually - we should not be seeing a semantic difference between no properties (as a NULL) and no properties as an empty string. Where we are making that distinction IMHO needs rethinking.

Reply:
This is in the low level **internal** documentation.

I figured that converting _NULL_ to **""** was worthwhile before this stage -- it's definitely more efficient.  That's what this PR is about.  I can move the conversions down to this level but it requires more knowledge about some of the callback structures than I'd like to expose.

Reply:
Essentially, this is the path of least resistance.

Reply:
This seems ok to me.

Reply:
Since the pattern is
if (!cache_get(propq))
   cache_set(propq);
   
Since doing conversions inside each of these calls as well as potentially in other calls before the set, it seems like setting in beforehand is a better option.